pkg/apis/eventing/v1alpha1/channel_types.go

```diff
@@ -139,11 +143,28 @@ func (cs *ChannelStatus) InitializeConditions() {
 // MarkProvisioned sets ChannelConditionProvisioned condition to True state.
 func (cs *ChannelStatus) MarkProvisioned() {
 	chanCondSet.Manage(cs).MarkTrue(ChannelConditionProvisioned)
+
+	// Channel-default-controller sets ChannelConditionProvisionerInstalled=False, and it needs to be set to True by individual controllers
+	// This is done so that each individual channel controller gets it for free.
+	cs.MarkProvisionerInstalled()
```

Comment:
I think the correct place to put this would be in `InitializeConditions()` at the end, add a call to `cs.MarkProvisionerInstalled()`

`InitializeConditions` is called on first pass reconciliation. It might be that provisioning gets blocked for an unknown reason and the better pattern is to let the controller "default" the `ProvisionerInstalled` condition on init, not on status change of something else. 

The way the code is currently you will likely get in a situation where the conditions are `unknown` (from the controller calling init on conditions) but still marked as  `ProvisionerInstalled=false`. But this would be a lie.

Reply:
Agree that provisioner could be present but due to some reason code got stuck or something and ProvisionerInstalled=False, which would misguide the user. 
However I don't agree that Initialize should only be called from the ProvisionerSpecificController.

How about Initialize(defaultController bool)? and set ProvisionerInstalled=True if defaultController=False?

When I was going over all conditions and reading them through kubectl on an object. I felt it to be useful to show all conditions and mark them as Unknown/True/False so that I know what is going on. Making the default controller call Initialize will make sure that that as soon as a Channel is created all its conditions are visible to the user.

Thoughts?